pkg/innerring/config.go

```diff
@@ -33,258 +24,189 @@ const (
 	}
 
 	return endpointsUnset, nil
 }
 
-func parseBlockchainConfig(v *viper.Viper, _logger *zap.Logger) (c blockchain.Config, err error) {
-	if !v.IsSet(cfgPathFSChainLocalConsensus) {
-		return c, fmt.Errorf("missing root section '%s'", cfgPathFSChainLocalConsensus)
+func validateBlockchainConfig(cfg *config.Config) error {
+	var err error
+	cfgFSChainLocalConsensus := &cfg.FSChain.Consensus
+
+	if !cfg.IsSet(cfgPathFSChainLocalConsensus) {
+		return fmt.Errorf("missing root section '%s'", cfgPathFSChainLocalConsensus)
 	}
 
-	_uint, err := parseConfigUint64Range(v, cfgPathFSChainLocalConsensus+".magic", "network magic", 1, math.MaxUint32)
-	if err != nil {
-		return c, err
+	magicKey := cfgPathFSChainLocalConsensus + ".magic"
+	if !cfg.IsSet(magicKey) {
+		return fmt.Errorf("missing section '%s'", magicKey)
+	}
+	if cfgFSChainLocalConsensus.Magic == 0 {
+		return fmt.Errorf("magic can't be zero '%s'", magicKey)
 	}
-	c.NetworkMagic = netmode.Magic(_uint)
 
-	var storageSection = cfgPathFSChainLocalConsensus + ".storage"
-	if !v.IsSet(storageSection) {
-		return c, fmt.Errorf("missing storage section '%s'", storageSection)
+	var (
+		storageSection = cfgPathFSChainLocalConsensus + ".storage"
+		storageTypeKey = storageSection + ".type"
+	)
+	if !cfg.IsSet(storageSection) {
+		return fmt.Errorf("missing storage section '%s'", storageSection)
 	}
-	var storageTypeKey = storageSection + ".type"
-	if !v.IsSet(storageTypeKey) {
-		return c, fmt.Errorf("missing storage type '%s'", storageTypeKey)
+	if !cfg.IsSet(storageTypeKey) {
+		return fmt.Errorf("missing storage type '%s'", storageTypeKey)
 	}
-	var storagePathKey = storageSection + ".path"
-	switch typ := v.GetString(storageTypeKey); typ {
-	default:
-		return c, fmt.Errorf("unsupported storage type '%s': '%s'", storageTypeKey, typ)
-	case dbconfig.BoltDB:
-		if !v.IsSet(storagePathKey) {
-			return c, fmt.Errorf("missing path to the BoltDB '%s'", storagePathKey)
-		}
-		c.Storage = blockchain.BoltDB(v.GetString(storagePathKey))
-	case dbconfig.LevelDB:
-		if !v.IsSet(storagePathKey) {
-			return c, fmt.Errorf("missing path to the LevelDB '%s'", storagePathKey)
-		}
-		c.Storage = blockchain.LevelDB(v.GetString(storagePathKey))
+	switch typ := cfgFSChainLocalConsensus.Storage.Type; typ {
 	case dbconfig.InMemoryDB:
-		c.Storage = blockchain.InMemory()
+	case dbconfig.BoltDB, dbconfig.LevelDB:
+		if cfgFSChainLocalConsensus.Storage.Path == "" {
+			return fmt.Errorf("missing path to the BoltDB '%s'", storageSection+".path")
+		}
+	default:
+		return fmt.Errorf("unsupported storage type '%s': '%s'", storageTypeKey, typ)
 	}
 
 	var committeeKey = cfgPathFSChainLocalConsensus + ".committee"
-	c.Committee, err = parseConfigPublicKeys(v, committeeKey, "committee members")
-	if err != nil {
-		return c, err
-	} else if len(c.Committee) == 0 {
-		return c, fmt.Errorf("empty committee members '%s'", committeeKey)
+	if !cfg.IsSet(committeeKey) {
```

Comment:
I think this duplicates the check below, if it's not set the length is going to be zero anyway.